apps/web/ee/lib/workflows/reminders/smsReminderManager.ts

```diff
@@ -0,0 +1,119 @@
+import {
+  WorkflowTriggerEvents,
+  TimeUnit,
+  WorkflowTemplates,
+  WorkflowActions,
+  WorkflowMethods,
+} from "@prisma/client/";
+import dayjs from "dayjs";
+import isBetween from "dayjs/plugin/isBetween";
+
+import * as twilio from "@ee/lib/workflows/reminders/smsProviders/twilioProvider";
+import smsReminderTemplate from "@ee/lib/workflows/reminders/templates/smsReminderTemplate";
+
+import prisma from "@lib/prisma";
+
+dayjs.extend(isBetween);
+
+export enum timeUnitLowerCase {
+  DAY = "day",
+  MINUTE = "minute",
+  YEAR = "year",
+}
+
+export type BookingInfo = {
+  uid?: string | null;
+  attendees: { name: string; email: string; timeZone: string }[];
+  organizer: { name: string; email: string };
+  startTime: string;
+  title: string;
+};
+
+export const scheduleSMSReminder = async (
+  evt: BookingInfo,
+  reminderPhone: string | null,
+  triggerEvent: WorkflowTriggerEvents,
+  action: WorkflowActions,
+  timeBefore: {
+    time: number | null;
+    timeUnit: TimeUnit | null;
+  },
+  message: string,
+  workflowStepId: number,
+  template: WorkflowTemplates
+) => {
+  const { startTime } = evt;
+  const uid = evt.uid as string;
+  const currentDate = dayjs();
+  const timeUnit: timeUnitLowerCase | undefined =
+    timeBefore.timeUnit?.toLocaleLowerCase() as timeUnitLowerCase;
+  const scheduledDate =
+    timeBefore.time && timeUnit ? dayjs(startTime).subtract(timeBefore.time, timeUnit) : null;
+
+  const name = action === WorkflowActions.SMS_ATTENDEE ? evt.attendees[0].name : "";
+  const attendeeName = action === WorkflowActions.SMS_ATTENDEE ? evt.organizer.name : evt.attendees[0].name;
+
+  switch (template) {
+    case WorkflowTemplates.REMINDER:
+      message =
+        smsReminderTemplate(evt.startTime, evt.title, evt.attendees[0].timeZone, attendeeName, name) ||
+        message;
+      break;
+  }
+
+  if (message.length > 0 && reminderPhone) {
+    //send SMS when event is booked/cancelled
+    if (
+      triggerEvent === WorkflowTriggerEvents.NEW_EVENT ||
+      triggerEvent === WorkflowTriggerEvents.EVENT_CANCELLED
+    ) {
+      try {
+        await twilio.sendSMS(reminderPhone, message);
+      } catch (error) {
+        console.log(`Error sending SMS with error ${error}`);
+      }
+    } else if (triggerEvent === WorkflowTriggerEvents.BEFORE_EVENT && scheduledDate) {
+      // Can only schedule at least 60 minutes in advance and at most 7 days in advance
+      if (
+        !currentDate.isBetween(scheduledDate.subtract(1, "hour"), scheduledDate) &&
+        scheduledDate.isBetween(currentDate, currentDate.add(7, "day"))
```

Comment:
Use `isAfter` instead like 
```suggestion
        scheduledDate.isAfter(currentDate.add(7, "days"))
```